storm-core/src/jvm/backtype/storm/validation/ConfigValidation.java

```diff
@@ -0,0 +1,518 @@
+            if (o == null) {
+                return;
+            }
+            //check if iterable
+            SimpleTypeValidator isIterable = new SimpleTypeValidator();
+            isIterable.validateField(name, Iterable.class, o);
+            for (Object entry : (Iterable) o) {
+                for (Class validator : validators) {
+                    Object v = validator.newInstance();
+                    if (v instanceof Validator) {
+                        ((Validator) v).validateField(name + " list entry", entry);
+                    } else {
+                        LOG.warn("validator: {} cannot be used in ListEntryCustomValidator.  Individual entry validators must a instance of Validator class", validator.getName());
+                    }
+                }
+            }
+
+        }
+    }
+
+    /**
+     * validates each key and value in a map of a certain type
+     */
+    public static class MapEntryTypeValidator {
+
+        public void validateField(String name, Class keyType, Class valueType, Object o) {
+            ConfigValidationUtils.NestableFieldValidator validator = ConfigValidationUtils.mapFv(keyType, valueType, false);
+            validator.validateField(name, o);
+        }
+    }
+
+    public static class MapEntryCustomValidator {
+
+        public void validateField(String name, Class[] keyValidators, Class[] valueValidators, Object o) throws IllegalAccessException, InstantiationException {
+            if (o == null) {
+                return;
+            }
+            //check if Map
+            SimpleTypeValidator isMap = new SimpleTypeValidator();
+            isMap.validateField(name, Map.class, o);
+            for (Map.Entry<Object, Object> entry : ((Map<Object, Object>) o).entrySet()) {
+                for (Class kv : keyValidators) {
+                    Object keyValidator = kv.newInstance();
+                    if (keyValidator instanceof Validator) {
+                        ((Validator) keyValidator).validateField(name + " Map key", entry.getKey());
+                    } else {
+                        LOG.warn("validator: {} cannot be used in MapEntryCustomValidator to validate keys.  Individual entry validators must a instance of Validator class", kv.getName());
+                    }
+                }
+                for (Class vv : valueValidators) {
+                    Object valueValidator = vv.newInstance();
+                    if (valueValidator instanceof Validator) {
+                        ((Validator) valueValidator).validateField(name + " Map value", entry.getValue());
+                    } else {
+                        LOG.warn("validator: {} cannot be used in MapEntryCustomValidator to validate values.  Individual entry validators must a instance of Validator class", vv.getName());
+                    }
+                }
+            }
+        }
+    }
```

Comment:
It might be good to add a concise javadoc for this class as we have done for others; maybe something like `validates each key and each value against the respective arrays of validators`.